lib/fs.js

```diff
@@ -1621,8 +1625,15 @@ function ReadStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedReadStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to ' +
+      'ReadStream is deprecated.',
+      warnedReadStreamOptions
+    );
+    options = options || {};
```

Comment:
@chrisdickinson +1.
We should know how behavior developers expect when 2nd argument is `function`.

And deprecation is not so good idea. This is not breaking change, we just followed the documentation.
https://iojs.org/api/fs.html#fs_fs_createreadstream_path_options

As a spec, we accept string and object only, `TypeError` is correct behavior.